app/status.go

```diff
@@ -21,174 +21,33 @@ func (a *App) GetUserStatusesByIds(userIDs []string) ([]*model.Status, *model.Ap
-		broadcast = true
-	} else {
-		if status.Manual && !manual {
-			return // manually set status always overrides non-manual one
-		}
-
-		if status.Status != model.StatusOnline {
-			broadcast = true
-		}
-
-		oldStatus = status.Status
-		oldTime = status.LastActivityAt
-		oldManual = status.Manual
-
-		status.Status = model.StatusOnline
-		status.Manual = false // for "online" there's no manual setting
-		status.LastActivityAt = model.GetMillis()
-	}
-
-	a.Srv().Platform().AddStatusCache(status)
-
-	// Only update the database if the status has changed, the status has been manually set,
-	// or enough time has passed since the previous action
-	if status.Status != oldStatus || status.Manual != oldManual || status.LastActivityAt-oldTime > model.StatusMinUpdateTime {
-		if broadcast {
-			if err := a.Srv().Store().Status().SaveOrUpdate(status); err != nil {
-				mlog.Warn("Failed to save status", mlog.String("user_id", userID), mlog.Err(err), mlog.String("user_id", userID))
-			}
-		} else {
-			if err := a.Srv().Store().Status().UpdateLastActivityAt(status.UserId, status.LastActivityAt); err != nil {
-				mlog.Error("Failed to save status", mlog.String("user_id", userID), mlog.Err(err), mlog.String("user_id", userID))
-			}
-		}
-	}
-
-	if broadcast {
-		a.Srv().Platform().BroadcastStatus(status)
-	}
+	a.Srv().Platform().SetStatusOnline(userID, manual)
 }
 
 func (a *App) SetStatusOffline(userID string, manual bool) {
-	if !*a.Config().ServiceSettings.EnableUserStatuses {
-		return
-	}
-
-	status, err := a.GetStatus(userID)
-	if err == nil && status.Manual && !manual {
-		return // manually set status always overrides non-manual one
-	}
-
-	status = &model.Status{UserId: userID, Status: model.StatusOffline, Manual: manual, LastActivityAt: model.GetMillis(), ActiveChannel: ""}
-
-	a.Srv().Platform().SaveAndBroadcastStatus(status)
+	a.Srv().Platform().SetStatusOffline(userID, manual)
 }
 
 func (a *App) SetStatusAwayIfNeeded(userID string, manual bool) {
-	if !*a.Config().ServiceSettings.EnableUserStatuses {
-		return
-	}
-
-	status, err := a.GetStatus(userID)
-
-	if err != nil {
-		status = &model.Status{UserId: userID, Status: model.StatusOffline, Manual: manual, LastActivityAt: 0, ActiveChannel: ""}
-	}
-
-	if !manual && status.Manual {
-		return // manually set status always overrides non-manual one
-	}
-
-	if !manual {
-		if status.Status == model.StatusAway {
-			return
-		}
-
-		if !a.IsUserAway(status.LastActivityAt) {
-			return
-		}
-	}
-
-	status.Status = model.StatusAway
-	status.Manual = manual
-	status.ActiveChannel = ""
-
-	a.Srv().Platform().SaveAndBroadcastStatus(status)
+	a.Srv().Platform().SetStatusAwayIfNeeded(userID, manual)
 }
 
 // SetStatusDoNotDisturbTimed takes endtime in unix epoch format in UTC
 // and sets status of given userId to dnd which will be restored back after endtime
 func (a *App) SetStatusDoNotDisturbTimed(userId string, endtime int64) {
```

Comment:
Do we still need these wrappers? Can't we call the platform methods directly?

Reply:
We can, there is no need to use wrappers here I guess. Need to check individually though. We should pick an approach to be consistent.

Reply:
I don't want to spend any more time on refactoring at this point. So let's leave this. This will become another never ending thing :P